securebanking-openbanking-uk-rcs-server/src/main/java/com/forgerock/securebanking/openbanking/uk/rcs/service/detail/FilePaymentConsentDetailsService.java

```diff
@@ -0,0 +1,84 @@
+import com.forgerock.securebanking.common.openbanking.uk.forgerock.datamodel.payment.FRAmount;
+import com.forgerock.securebanking.common.openbanking.uk.forgerock.datamodel.tpp.Tpp;
+import com.forgerock.securebanking.openbanking.uk.rcs.api.dto.consent.details.ConsentDetails;
+import com.forgerock.securebanking.openbanking.uk.rcs.api.dto.consent.details.FilePaymentConsentDetails;
+import com.forgerock.securebanking.openbanking.uk.rcs.client.idm.PaymentConsentService;
+import com.forgerock.securebanking.openbanking.uk.rcs.client.idm.TppService;
+import com.forgerock.securebanking.openbanking.uk.rcs.client.idm.dto.consent.FRFilePaymentConsent;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.org.openbanking.datamodel.payment.OBWriteDomestic2DataInitiationRemittanceInformation;
+import uk.org.openbanking.datamodel.payment.OBWriteFile2DataInitiation;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@Slf4j
+public class FilePaymentConsentDetailsService extends PaymentConsentDetailsService<FRFilePaymentConsent> {
+
+    private final PaymentConsentService paymentConsentService;
+
+    public FilePaymentConsentDetailsService(PaymentConsentService paymentConsentService, TppService tppService) {
+        super(paymentConsentService, tppService);
+        this.paymentConsentService = paymentConsentService;
+    }
+
+    @Override
+    protected FRFilePaymentConsent getConsent(String consentId) {
+        log.debug("Retrieving file payment consent with ID {}", consentId);
+        return paymentConsentService.getConsent(consentId, FRFilePaymentConsent.class);
+    }
+
+    @Override
+    protected String getDebitAccountId(FRFilePaymentConsent paymentConsent) {
+        OBWriteFile2DataInitiation initiation = paymentConsent.getData().getInitiation();
+        if (initiation.getDebtorAccount() == null) {
+            return null;
+        }
+        return initiation.getDebtorAccount().getIdentification();
+    }
+
+    @Override
+    protected ConsentDetails buildResponse(FRFilePaymentConsent paymentConsent,
+                                           List<FRAccountWithBalance> accounts,
+                                           Tpp tpp) {
+        OBWriteFile2DataInitiation initiation = paymentConsent.getData().getInitiation();
+        return FilePaymentConsentDetails.builder()
+                .accounts(accounts)
+                .username(paymentConsent.getResourceOwnerUsername())
+                .clientId(tpp.getClientId())
+                .merchantName(paymentConsent.getOauth2ClientName())
+                .fileReference(initiation.getFileReference())
+                .numberOfTransactions(initiation.getNumberOfTransactions())
+                .totalAmount(FRAmount.builder()
+                        .amount(initiation.getControlSum().toPlainString())
+                        // TODO - will this always be GBP?
+                        .currency("GBP")
```

Comment:
The currency should not come from the consent or payment initiation? I think that isn't correct to have a hardcoded  value.

Reply:
No it doesn't come from the consent or the payment initiation, which is why I've added the TODO. I've been discussing this with Christian and it's something we might need to look at further, but I haven't tested the UI with file payments yet. This is one of the flows we'll have to test once everything is hooked up.